src/test/java/org/apache/sysds/test/functions/privacy/FederatedL2SVMTest.java

```diff
@@ -0,0 +1,386 @@
+		federatedL2SVM(Types.ExecMode.SINGLE_NODE, privacyConstraints, null, PrivacyLevel.Private, false, null, true, DMLException.class);
+	}
+
+	@Test
+	public void federatedL2SVMCPPrivatePrivateAggregationFederatedYX1() throws JSONException {
+		Map<String, PrivacyConstraint> privacyConstraints = new HashMap<>();
+		privacyConstraints.put("Y", new PrivacyConstraint(PrivacyLevel.Private));
+		privacyConstraints.put("X1", new PrivacyConstraint(PrivacyLevel.PrivateAggregation));
+		federatedL2SVMNoException(Types.ExecMode.SINGLE_NODE, privacyConstraints, null, PrivacyLevel.Private);
+	}
+
+	@Test
+	public void federatedL2SVMCPPrivatePrivateAggregationFederatedYX2() throws JSONException {
+		Map<String, PrivacyConstraint> privacyConstraints = new HashMap<>();
+		privacyConstraints.put("Y", new PrivacyConstraint(PrivacyLevel.Private));
+		privacyConstraints.put("X2", new PrivacyConstraint(PrivacyLevel.PrivateAggregation));
+		federatedL2SVMNoException(Types.ExecMode.SINGLE_NODE, privacyConstraints, null, PrivacyLevel.Private);
+	}
+
+	@Test
+	public void federatedL2SVMCPPrivatePrivateAggregationFederatedX2X1() throws JSONException {
+		Map<String, PrivacyConstraint> privacyConstraints = new HashMap<>();
+		privacyConstraints.put("X2", new PrivacyConstraint(PrivacyLevel.Private));
+		privacyConstraints.put("X1", new PrivacyConstraint(PrivacyLevel.PrivateAggregation));
+		federatedL2SVM(Types.ExecMode.SINGLE_NODE, privacyConstraints, null, PrivacyLevel.Private, false, null, true, DMLException.class);
+	}
+
+	// Require Federated Workers to return matrix
+
+	@Test
+	public void federatedL2SVMCPPrivateAggregationX1Exception() throws JSONException {
+		this.rows = 1000;
+		this.cols = 1;
+		Map<String, PrivacyConstraint> privacyConstraints = new HashMap<>();
+		privacyConstraints.put("X1", new PrivacyConstraint(PrivacyLevel.PrivateAggregation));
+		federatedL2SVM(Types.ExecMode.SINGLE_NODE, privacyConstraints, null, PrivacyLevel.PrivateAggregation, false, null, true, DMLException.class);
+	}
+
+	@Test
+	public void federatedL2SVMCPPrivateAggregationX2Exception() throws JSONException {
+		this.rows = 1000;
+		this.cols = 1;
+		Map<String, PrivacyConstraint> privacyConstraints = new HashMap<>();
+		privacyConstraints.put("X2", new PrivacyConstraint(PrivacyLevel.PrivateAggregation));
+		federatedL2SVM(Types.ExecMode.SINGLE_NODE, privacyConstraints, null, PrivacyLevel.PrivateAggregation, false, null, true, DMLException.class);
+	}
+
+	@Test
+	public void federatedL2SVMCPPrivateX1Exception() throws JSONException {
+		this.rows = 1000;
+		this.cols = 1;
+		Map<String, PrivacyConstraint> privacyConstraints = new HashMap<>();
+		privacyConstraints.put("X1", new PrivacyConstraint(PrivacyLevel.Private));
+		federatedL2SVM(Types.ExecMode.SINGLE_NODE, privacyConstraints, null, PrivacyLevel.Private, false, null, true, DMLException.class);
+	}
+
+	@Test
+	public void federatedL2SVMCPPrivateX2Exception() throws JSONException {
```

Comment:
Every method in this class throws JSONException. Can this be caught, since i worry that if for some reason it is thrown we get the odd error of fork crashed in our tests.?

Reply:
Are you sure that it will crash? When I run it locally, the test will just fail and it will print the exception. This is easier to work with compared to if the exception is caught.